prog/vm/github_runner.rb

```diff
@@ -200,6 +201,23 @@ def quota_available?
       Clog.emit("allowed because of low utilization") { {exceeded_concurrency_limit: {family_utilization:, label: github_runner.label, repository_name: github_runner.repository_name}} }
     end
     github_runner.log_duration("runner_capacity_waited", Time.now - github_runner.created_at)
+
+    hop_apply_custom_label_quota if GithubCustomLabel.where(label: github_runner.label, installation_id: github_runner.installation_id).any?
+    hop_allocate_vm
+  end
+
+  label def apply_custom_label_quota
+    label = github_runner.label
+    installation_id = github_runner.installation_id
+    custom_label = GithubCustomLabel.where(label:, installation_id:).first
+    hop_allocate_vm unless custom_label
+
+    unless custom_label.concurrency_limit_available?
+      Clog.emit("hit custom label concurrency limit") { {custom_label_concurrency_limit: {label:, installation_id:, repository_name: github_runner.repository_name}} }
+      nap rand(5..15)
+    end
+
+    custom_label.update(allocated_runner_count: custom_label.allocated_runner_count + 1)
```

Comment:
You can do this with a single DB query, which is also more concurrency-efficient. Something like:
```suggestion
    new_runner_count = Sequel[:allocated_runner_count] + 1
    updated_rows = GithubCustomLabel
      .where(label:, installation_id:)
      .where(new_runner_count <= concurrent_runner_count_limit)
      .update(allocated_runner_count: new_runner_count)

    if updated_rows != 1
      Clog.emit("hit custom label concurrency limit") { {custom_label_concurrency_limit: {label:, installation_id:, repository_name: github_runner.repository_name}} }
      nap rand(5..15)
    end
```